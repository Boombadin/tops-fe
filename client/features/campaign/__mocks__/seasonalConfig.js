export default {
  sku: ['123', '456', '789'],
  remark: {
    th: {
      homepage: 'หมายเหตุหน้าโฮม',
    },
    en: {
      homepage: 'homepage remark',
    },
  },
  style: {
    web: {
      border_color: '#f70405',
      homepage_remark_font_color: '#f70405',
      banner: {
        head_title_header_icon_url: 'https://i.ibb.co/0V4yL9f/group-11-3x.png',
        tail_title_header_icon_url:
          'https://i.ibb.co/h8cwn5n/china-lamp-right-3x.png',
        text_link_font_color: '#FFFFFF',
        text_link_url: 'https://www.google.com',
        title_header_font_color: '#ffdb71',
        th: {
          banner_image_url: 'https://i.ibb.co/2Y4jMKS/f40405.png',
          text_link: 'แสดงทั้งหมด',
          title_header: 'เทศกาลวันสาร์สจีน',
        },
        en: {
          banner_image_url: 'https://i.ibb.co/2Y4jMKS/f40405.png',
          text_link: 'Show All',
          title_header: 'Chinese Ghost Festival',
        },
      },
    },
  },
};
