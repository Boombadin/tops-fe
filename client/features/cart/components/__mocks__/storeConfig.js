export const mockStoreConfig = JSON.parse(
  '{"id":17,"code":"tops_sa_432_en","website_id":9,"locale":"en_US","base_currency_code":"THB","default_display_currency_code":"THB","timezone":"Asia/Bangkok","weight_unit":"kgs","base_url":"https://uat-mdc.tops.co.th/","base_link_url":"https://uat-mdc.tops.co.th/tops_sa_432_en/","base_static_url":"https://uat-mdc.tops.co.th/static/version1603351164/","base_media_url":"https://uat-mdc.tops.co.th/media/","secure_base_url":"https://uat-mdc.tops.co.th/","secure_base_link_url":"https://uat-mdc.tops.co.th/tops_sa_432_en/","secure_base_static_url":"https://uat-mdc.tops.co.th/static/version1603351164/","secure_base_media_url":"https://uat-mdc.tops.co.th/media/","extension_attributes":{"grab_url":"https://partner-api.stg-myteksi.com/","grab_client_id":"MzE5MmYwOTA4MDM4NDU0NWI1NGM3NDhjNTFkNWUxM2U=","fe_url":"https://uat.tops.co.th/en/","b1gv_name":"Buy 2 Pcs. {{price}}฿ (Save {{promotion_amount}}฿ )","b1gv_image":"promotion_badges/stores/17/b1gv.png","b2gv_name":"Buy 3 Pcs. {{price}}฿ (Save {{promotion_amount}}฿ )","b2gv_image":"promotion_badges/stores/17/b2gv.png","b3gv_name":"Buy 4 Pcs. {{price}} (Save {{promotion_amount}} )","b3gv_image":"promotion_badges/stores/17/b3gv.png","bogo_name":"Buy 2 Pay 1","bogo_image":"promotion_badges/stores/17/bogo.png","b2g1_name":"Buy 3 Pay 2","b2g1_image":"promotion_badges/stores/17/b2g1.png","b3g1_name":"Buy 4 Pay 3","b3g1_image":"promotion_badges/stores/17/b3g1.png","redhot_name":"Save {{promotion_amount}} THB.","redhot_image":"promotion_badges/stores/17/redhot.png","sale_name":"Save {{promotion_amount}} THB.","sale_image":"promotion_badges/websites/9/sale.png","moneyback_name":"Get {{the1card_point}} extra points / {{the1card_qty}} {{consumer_unit}}","moneyback_image":"promotion_badges/stores/17/moneyback.png","branch_io_key":"key_test_mnFGH7nF2yl9DCVzTclklkbmxsenhP0z"}}',
);
