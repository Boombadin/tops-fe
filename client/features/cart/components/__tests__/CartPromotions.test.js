import { render } from '@testing-library/react';
import React from 'react';

import { mockItem, mockProductBundle } from '../__mocks__/productBundle';
import { mockStoreConfig } from '../__mocks__/storeConfig';
import CartPromotions from '../CartPromotions';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});
jest.mock('@client/contexts', () => ({
  useCartContext: jest.fn().mockReturnValue({
    cartAction: {
      deleteProductBundle: jest.fn(),
    },
  }),
}));

describe('features/cart/components/CartPromotion', () => {
  const initialState = {
    items: mockItem,
    itemIds: [
      {
        item_id: 11577888,
        item_sku: '8850187400030',
      },
      {
        item_id: 11577888,
        item_sku: '8850187400030',
      },
    ],
    discountAmount: '20.00',
    promoQtyStep: 2,
    actionDiscountType: 'groupn_fixdisc',
    isMiniCart: false,
    bundle: mockProductBundle,
    storeConfig: mockStoreConfig,
    deleteBundleItem: () => {},
    handlerViewProduct: () => {},
    openPromoBundleModal: () => {},
  };

  test(`it cart promotion desktop should match with snapshot`, () => {
    const { asFragment } = render(<CartPromotions {...initialState} />);
    expect(asFragment(<CartPromotions {...initialState} />)).toMatchSnapshot();
  });

  test(`it cart promotion mobile should match with snapshot`, () => {
    const initialState = {
      ...initialState,
      isMiniCart: true,
    };
    const { asFragment } = render(<CartPromotions {...initialState} />);
    expect(asFragment(<CartPromotions {...initialState} />)).toMatchSnapshot();
  });

  test(`it cart promotion when cunsumerUnit not empty should match with snapshot`, () => {
    const initialState = {
      ...initialState,
      item: {
        ...initialState?.item,
        weight_item_ind: '1',
        selling_unit: {
          ...initialState?.item?.selling_unit,
          consumerUnit: 'kg',
        },
      },
    };
    const { asFragment } = render(<CartPromotions {...initialState} />);
    expect(asFragment(<CartPromotions {...initialState} />)).toMatchSnapshot();
  });

  test(`it cart promotion when store TH should match with snapshot`, () => {
    const initialState = {
      ...initialState,
      lang: {
        code: 'th_TH',
      },
    };
    const { asFragment } = render(<CartPromotions {...initialState} />);
    expect(asFragment(<CartPromotions {...initialState} />)).toMatchSnapshot();
  });
});
