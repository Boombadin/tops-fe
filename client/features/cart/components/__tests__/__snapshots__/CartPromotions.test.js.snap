// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/cart/components/CartPromotion it cart promotion desktop should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="Media__HideMobile-sc-1gp5uia-0 bYPzTW"
  >
    <div
      class="CartDesktopPromotion__RowContainerItem-sc-1tma7-3 djifYf Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
    >
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 drldVy"
      >
        <div
          class="Row__Wrapper-v6uxgu-0 kSLyDU"
        >
          <div
            class="CartDesktopPromotion__ColPromotion-sc-1tma7-6 dBVfNp Col__Column-sc-1619uro-0 bpIQSr"
          >
            <p
              class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 knxMnr"
              color="#666666"
              display=""
              size="11"
              spacing="normal"
            >
              promotion.promotion_online
            </p>
          </div>
        </div>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 djatvc"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 bUWYUb"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 djatvc"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 gIiNII"
      >
        <p
          class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 cwWbaV"
          color="#ec1d24"
          display=""
          size="11"
          spacing="normal"
        >
          -20.00฿
        </p>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 iZdHvi"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 cOPlex"
      >
        <img
          class="Icon-sc-134752z-0 CartDesktopPromotion__IconImage-sc-1tma7-5 jgMlek"
          data-testid="btn-removeCartDesktopPromotion-onDeleteItem"
          height="15"
          opacity="1"
          rotate="0"
          src="/assets/icons/round-delete-24-px.svg"
          width="11.7"
        />
      </div>
    </div>
  </div>
  <div
    class="Media__HideDesktop-sc-1gp5uia-1 jUPJmB"
  >
    <div
      class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 dTlSQg Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
    >
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 dSStdr Col__Column-sc-1619uro-0 drldVy"
      >
        <div
          class="Row__Wrapper-v6uxgu-0 kSLyDU"
        >
          <div
            class="CartMobilePromotion__ColPromotion-sc-1e93wxj-6 fXheif Col__Column-sc-1619uro-0 bpIQSr"
          >
            <p
              class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 fNPGig"
              color="#666666"
              display=""
              size="11"
              spacing="normal"
            >
              promotion.promotion_online
            </p>
          </div>
        </div>
      </div>
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jzEWsS Col__Column-sc-1619uro-0 fZZAYg"
      >
        <div
          class="Row__Wrapper-v6uxgu-0 kSLyDU"
        >
          <div
            class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 lbkIwJ Col__Column-sc-1619uro-0 jnKyiD"
          >
            <p
              class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 cwWbaV"
              color="#ec1d24"
              display=""
              size="11"
              spacing="normal"
            >
              -20.00฿
            </p>
          </div>
          <div
            class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jKZIkT Col__Column-sc-1619uro-0 dpfeHD"
          >
            <img
              class="Icon-sc-134752z-0 CartMobilePromotion__IconImage-sc-1e93wxj-3 dVSXfd"
              data-testid="btn-removeCartMobilePromotion-onDeleteItem"
              height="15"
              opacity="1"
              rotate="0"
              src="/assets/icons/round-delete-24-px.svg"
              width="11.7"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/cart/components/CartPromotion it cart promotion mobile should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 hqydJY Row__Wrapper-v6uxgu-0 kSLyDU"
    color="#f7f7f7"
  >
    <div
      class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 jifnjv"
    >
      <img
        class="CartMobilePromotion__ImageProduct-sc-1e93wxj-4 caSzbV Image__Img-sc-1s8ofg8-0 bEmjdg"
        data-testid="img-viewCartMobilePromotion-onViewProduct-"
        src="catalog/product?imwidth=50"
        width="40px"
      />
    </div>
    <div
      class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 drldVy"
    >
      <div
        class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 dTlSQg Row__Wrapper-v6uxgu-0 kSLyDU"
        color="#f7f7f7"
      >
        <div
          class="Col__Column-sc-1619uro-0 bpIQSr"
        >
          <div
            class="Padding-sc-10o2jvk-0 GtPWG"
          >
            <div
              class="Row__Wrapper-v6uxgu-0 kSLyDU"
            >
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jJXAgC Col__Column-sc-1619uro-0 dYSUbG"
              >
                <span
                  class="Text-sc-9p67zt-0 CartMobilePromotion__Text2LineEllipsis-sc-1e93wxj-2 gnuuoF"
                  color="#666666"
                  data-testid="btn-viewCartMobilePromotion-onViewProduct-"
                  display=""
                  id="txt-product-name"
                  size="11"
                  spacing="normal"
                  style="font-family: sans-serif;"
                >
                  promotion.free: 
                </span>
              </div>
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 kSzlfq Col__Column-sc-1619uro-0 ecpbdq"
              >
                <span
                  class="Text-sc-9p67zt-0 iqhbTf"
                  color="#666666"
                  display=""
                  size="11"
                  spacing="normal"
                >
                  0 
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Col__Column-sc-1619uro-0 bpIQSr"
        >
          <div
            class="Row__Wrapper-v6uxgu-0 kSLyDU"
          >
            <div
              class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 dSStdr Col__Column-sc-1619uro-0 drldVy"
            >
              <span
                class="Text-sc-9p67zt-0 CartMobilePromotion__TextChangeBundle-sc-1e93wxj-5 jpooFB"
                color="#666666"
                data-testid="btn-viewCartMobilePromotion-openPromoBundleModal"
                display=""
                size="10"
                spacing="normal"
              >
                <img
                  class="Icon-sc-134752z-0 fCCZuM"
                  height="0"
                  opacity="1"
                  rotate="0"
                  src="/assets/icons/ic-edit.svg"
                  style="margin-right: 3px;"
                  width="10"
                />
                <p
                  class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 khELOm"
                  color="#666666"
                  display=""
                  size="11"
                  spacing="normal"
                >
                  promotion.change_free_items
                </p>
              </span>
            </div>
            <div
              class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jzEWsS Col__Column-sc-1619uro-0 fZZAYg"
            >
              <div
                class="Row__Wrapper-v6uxgu-0 kSLyDU"
              >
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 lbkIwJ Col__Column-sc-1619uro-0 jnKyiD"
                >
                  <span
                    class="Text-sc-9p67zt-0 ipmWsm"
                    color="#666666"
                    display=""
                    size="12"
                    spacing="normal"
                  >
                    promotion.free
                  </span>
                </div>
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jKZIkT Col__Column-sc-1619uro-0 dpfeHD"
                >
                  <img
                    class="Icon-sc-134752z-0 CartMobilePromotion__IconImage-sc-1e93wxj-3 dVSXfd"
                    data-testid="btn-removeCartMobilePromotion-onDeleteItem"
                    height="15"
                    opacity="1"
                    rotate="0"
                    src="/assets/icons/round-delete-24-px.svg"
                    width="11.7"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/cart/components/CartPromotion it cart promotion when cunsumerUnit not empty should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="Media__HideMobile-sc-1gp5uia-0 bYPzTW"
  >
    <div
      class="CartDesktopPromotion__RowContainerItem-sc-1tma7-3 dLMZCW Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
      height="60px"
    >
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 drldVy"
      >
        <img
          class="CartDesktopPromotion__ImageProduct-sc-1tma7-0 jqfqJZ Image__Img-sc-1s8ofg8-0 bEmjdg"
          data-testid="img-viewCartDesktopPromotion-onViewProduct-"
          src="catalog/product?imwidth=50"
          width="40px"
        />
        <div
          class="Row__Wrapper-v6uxgu-0 kSLyDU"
        >
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <span
              class="Text-sc-9p67zt-0 CartDesktopPromotion__Text2LineEllipsis-sc-1tma7-2 bwUQTO"
              color="#666666"
              data-testid="btn-viewCartDesktopPromotion-onViewProduct-"
              display=""
              id="txt-product-name"
              size="12"
              spacing="normal"
            />
          </div>
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <span
              class="Text-sc-9p67zt-0 CartDesktopPromotion__TextChangeBundle-sc-1tma7-1 blczQl"
              color="#666666"
              data-testid="btn-viewCartDesktopPromotion-openPromoBundleModal"
              display=""
              size="10"
              spacing="normal"
            >
              <img
                class="Icon-sc-134752z-0 fCCZuM"
                height="0"
                opacity="1"
                rotate="0"
                src="/assets/icons/ic-edit.svg"
                style="margin-right: 1px;"
                width="10"
              />
              <p
                class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 khELOm"
                color="#666666"
                display=""
                size="11"
                spacing="normal"
                style="font-size: 10px;"
              >
                promotion.change_free_items
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 djatvc"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        />
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 bUWYUb"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        />
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 djatvc"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        >
          kg
        </span>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 gIiNII"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 iZdHvi"
      >
        <span
          class="Text-sc-9p67zt-0 bjEjky"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        >
          promotion.free
        </span>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 cOPlex"
      >
        <img
          class="Icon-sc-134752z-0 CartDesktopPromotion__IconImage-sc-1tma7-5 jgMlek"
          data-testid="btn-removeCartDesktopPromotion-onDeleteItem"
          height="15"
          opacity="1"
          rotate="0"
          src="/assets/icons/round-delete-24-px.svg"
          width="11.7"
        />
      </div>
    </div>
  </div>
  <div
    class="Media__HideDesktop-sc-1gp5uia-1 jUPJmB"
  >
    <div
      class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 hqydJY Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
    >
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 jifnjv"
      >
        <img
          class="CartMobilePromotion__ImageProduct-sc-1e93wxj-4 caSzbV Image__Img-sc-1s8ofg8-0 bEmjdg"
          data-testid="img-viewCartMobilePromotion-onViewProduct-"
          src="catalog/product?imwidth=50"
          width="40px"
        />
      </div>
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 drldVy"
      >
        <div
          class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 dTlSQg Row__Wrapper-v6uxgu-0 kSLyDU"
          color="#f7f7f7"
        >
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <div
              class="Padding-sc-10o2jvk-0 GtPWG"
            >
              <div
                class="Row__Wrapper-v6uxgu-0 kSLyDU"
              >
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jJXAgC Col__Column-sc-1619uro-0 dYSUbG"
                >
                  <span
                    class="Text-sc-9p67zt-0 CartMobilePromotion__Text2LineEllipsis-sc-1e93wxj-2 gnuuoF"
                    color="#666666"
                    data-testid="btn-viewCartMobilePromotion-onViewProduct-"
                    display=""
                    id="txt-product-name"
                    size="11"
                    spacing="normal"
                    style="font-family: sans-serif;"
                  >
                    promotion.free: 
                  </span>
                </div>
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 kSzlfq Col__Column-sc-1619uro-0 ecpbdq"
                >
                  <span
                    class="Text-sc-9p67zt-0 iqhbTf"
                    color="#666666"
                    display=""
                    size="11"
                    spacing="normal"
                  >
                    0 kg
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <div
              class="Row__Wrapper-v6uxgu-0 kSLyDU"
            >
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 dSStdr Col__Column-sc-1619uro-0 drldVy"
              >
                <span
                  class="Text-sc-9p67zt-0 CartMobilePromotion__TextChangeBundle-sc-1e93wxj-5 jpooFB"
                  color="#666666"
                  data-testid="btn-viewCartMobilePromotion-openPromoBundleModal"
                  display=""
                  size="10"
                  spacing="normal"
                >
                  <img
                    class="Icon-sc-134752z-0 fCCZuM"
                    height="0"
                    opacity="1"
                    rotate="0"
                    src="/assets/icons/ic-edit.svg"
                    style="margin-right: 3px;"
                    width="10"
                  />
                  <p
                    class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 khELOm"
                    color="#666666"
                    display=""
                    size="11"
                    spacing="normal"
                  >
                    promotion.change_free_items
                  </p>
                </span>
              </div>
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jzEWsS Col__Column-sc-1619uro-0 fZZAYg"
              >
                <div
                  class="Row__Wrapper-v6uxgu-0 kSLyDU"
                >
                  <div
                    class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 lbkIwJ Col__Column-sc-1619uro-0 jnKyiD"
                  >
                    <span
                      class="Text-sc-9p67zt-0 ipmWsm"
                      color="#666666"
                      display=""
                      size="12"
                      spacing="normal"
                    >
                      promotion.free
                    </span>
                  </div>
                  <div
                    class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jKZIkT Col__Column-sc-1619uro-0 dpfeHD"
                  >
                    <img
                      class="Icon-sc-134752z-0 CartMobilePromotion__IconImage-sc-1e93wxj-3 dVSXfd"
                      data-testid="btn-removeCartMobilePromotion-onDeleteItem"
                      height="15"
                      opacity="1"
                      rotate="0"
                      src="/assets/icons/round-delete-24-px.svg"
                      width="11.7"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/cart/components/CartPromotion it cart promotion when store TH should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="Media__HideMobile-sc-1gp5uia-0 bYPzTW"
  >
    <div
      class="CartDesktopPromotion__RowContainerItem-sc-1tma7-3 dLMZCW Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
      height="60px"
    >
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 drldVy"
      >
        <img
          class="CartDesktopPromotion__ImageProduct-sc-1tma7-0 jqfqJZ Image__Img-sc-1s8ofg8-0 bEmjdg"
          data-testid="img-viewCartDesktopPromotion-onViewProduct-"
          src="catalog/product?imwidth=50"
          width="40px"
        />
        <div
          class="Row__Wrapper-v6uxgu-0 kSLyDU"
        >
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <span
              class="Text-sc-9p67zt-0 CartDesktopPromotion__Text2LineEllipsis-sc-1tma7-2 bwUQTO"
              color="#666666"
              data-testid="btn-viewCartDesktopPromotion-onViewProduct-"
              display=""
              id="txt-product-name"
              size="12"
              spacing="normal"
            />
          </div>
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <span
              class="Text-sc-9p67zt-0 CartDesktopPromotion__TextChangeBundle-sc-1tma7-1 blczQl"
              color="#666666"
              data-testid="btn-viewCartDesktopPromotion-openPromoBundleModal"
              display=""
              size="10"
              spacing="normal"
            >
              <img
                class="Icon-sc-134752z-0 fCCZuM"
                height="0"
                opacity="1"
                rotate="0"
                src="/assets/icons/ic-edit.svg"
                style="margin-right: 1px;"
                width="10"
              />
              <p
                class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 khELOm"
                color="#666666"
                display=""
                size="11"
                spacing="normal"
                style="font-size: 10px;"
              >
                promotion.change_free_items
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 djatvc"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        />
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 bUWYUb"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        />
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 kHbetE Col__Column-sc-1619uro-0 djatvc"
      >
        <span
          class="Text-sc-9p67zt-0 hYkHxI"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        />
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 gIiNII"
      />
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 yUtop Col__Column-sc-1619uro-0 iZdHvi"
      >
        <span
          class="Text-sc-9p67zt-0 bjEjky"
          color="#666666"
          display=""
          size="12"
          spacing="normal"
        >
          promotion.free
        </span>
      </div>
      <div
        class="CartDesktopPromotion__ColumnItem-sc-1tma7-4 iLtDlt Col__Column-sc-1619uro-0 cOPlex"
      >
        <img
          class="Icon-sc-134752z-0 CartDesktopPromotion__IconImage-sc-1tma7-5 jgMlek"
          data-testid="btn-removeCartDesktopPromotion-onDeleteItem"
          height="15"
          opacity="1"
          rotate="0"
          src="/assets/icons/round-delete-24-px.svg"
          width="11.7"
        />
      </div>
    </div>
  </div>
  <div
    class="Media__HideDesktop-sc-1gp5uia-1 jUPJmB"
  >
    <div
      class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 hqydJY Row__Wrapper-v6uxgu-0 kSLyDU"
      color="#f7f7f7"
    >
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 jifnjv"
      >
        <img
          class="CartMobilePromotion__ImageProduct-sc-1e93wxj-4 caSzbV Image__Img-sc-1s8ofg8-0 bEmjdg"
          data-testid="img-viewCartMobilePromotion-onViewProduct-"
          src="catalog/product?imwidth=50"
          width="40px"
        />
      </div>
      <div
        class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jtiPwC Col__Column-sc-1619uro-0 drldVy"
      >
        <div
          class="CartMobilePromotion__RowContainerItem-sc-1e93wxj-0 dTlSQg Row__Wrapper-v6uxgu-0 kSLyDU"
          color="#f7f7f7"
        >
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <div
              class="Padding-sc-10o2jvk-0 GtPWG"
            >
              <div
                class="Row__Wrapper-v6uxgu-0 kSLyDU"
              >
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jJXAgC Col__Column-sc-1619uro-0 dYSUbG"
                >
                  <span
                    class="Text-sc-9p67zt-0 CartMobilePromotion__Text2LineEllipsis-sc-1e93wxj-2 gnuuoF"
                    color="#666666"
                    data-testid="btn-viewCartMobilePromotion-onViewProduct-"
                    display=""
                    id="txt-product-name"
                    size="11"
                    spacing="normal"
                    style="font-family: sans-serif;"
                  >
                    promotion.free: 
                  </span>
                </div>
                <div
                  class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 kSzlfq Col__Column-sc-1619uro-0 ecpbdq"
                >
                  <span
                    class="Text-sc-9p67zt-0 iqhbTf"
                    color="#666666"
                    display=""
                    size="11"
                    spacing="normal"
                  >
                    0 
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Col__Column-sc-1619uro-0 bpIQSr"
          >
            <div
              class="Row__Wrapper-v6uxgu-0 kSLyDU"
            >
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 dSStdr Col__Column-sc-1619uro-0 drldVy"
              >
                <span
                  class="Text-sc-9p67zt-0 CartMobilePromotion__TextChangeBundle-sc-1e93wxj-5 jpooFB"
                  color="#666666"
                  data-testid="btn-viewCartMobilePromotion-openPromoBundleModal"
                  display=""
                  size="10"
                  spacing="normal"
                >
                  <img
                    class="Icon-sc-134752z-0 fCCZuM"
                    height="0"
                    opacity="1"
                    rotate="0"
                    src="/assets/icons/ic-edit.svg"
                    style="margin-right: 3px;"
                    width="10"
                  />
                  <p
                    class="Text-sc-9p67zt-0 default-font TextGuide__Caption-sc-1wuo3jj-0 khELOm"
                    color="#666666"
                    display=""
                    size="11"
                    spacing="normal"
                  >
                    promotion.change_free_items
                  </p>
                </span>
              </div>
              <div
                class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jzEWsS Col__Column-sc-1619uro-0 fZZAYg"
              >
                <div
                  class="Row__Wrapper-v6uxgu-0 kSLyDU"
                >
                  <div
                    class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 lbkIwJ Col__Column-sc-1619uro-0 jnKyiD"
                  >
                    <span
                      class="Text-sc-9p67zt-0 ipmWsm"
                      color="#666666"
                      display=""
                      size="12"
                      spacing="normal"
                    >
                      promotion.free
                    </span>
                  </div>
                  <div
                    class="CartMobilePromotion__ColumnItem-sc-1e93wxj-1 jKZIkT Col__Column-sc-1619uro-0 dpfeHD"
                  >
                    <img
                      class="Icon-sc-134752z-0 CartMobilePromotion__IconImage-sc-1e93wxj-3 dVSXfd"
                      data-testid="btn-removeCartMobilePromotion-onDeleteItem"
                      height="15"
                      opacity="1"
                      rotate="0"
                      src="/assets/icons/round-delete-24-px.svg"
                      width="11.7"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
